### PR TITLE
Preserving canceled bids in Bids collection instead of bids-archive

### PR DESCRIPTION
**Pull request by Will-Hyde**
This change should begin marking canceled bids with a cancel date and preserving them in the main bids collection, rather than moving them to the respective archive collection

### CODE REVIEW

File: src/database/index.ts [hunk 1]
```diff
@@ -6,6 +6,6 @@ export interface BidDatabaseService {
   insertBids: (data: Bid[]) => Promise<boolean>;
   getBidsByTokenIds: (nftIds: string[], bidStatus: BidFilterStatus) => Promise<Bid[]>;
   getBidsByAccount: (account: string, bidStatus: BidFilterStatus) => Promise<Bid[]>;
-  getBidBySignedMessage: (signedMessage: string) => Promise<Bid | null>;
-  cancelBid: (bid: CancelableBid, archiveCollection: string) => Promise<boolean>;
+  getBidBySignedMessage: (signedMessage: string) => Promise<CancelableBid | null>;
```
Comment (JamesEarle) [inline src/database/index.ts:9]: This being a `CancelableBid` conceptually is a bit weird to me. Every bid is cancelable so should every bid be a `CancelableBid`? Should the above `insertBids` function take in a `CancelableBid[]`? 

I mentioned in your other PR that having a `CanceledBid` interface would be preferable and in the call to `cancelBid` we could take in a `Bid` and then transform to a `CanceledBid` in that call. That would keep the concept of cancelling a bid isolated to only functions that use it

**Review by JamesEarle — APPROVED** [review]
LGTM but a comment about the `CancelableBid` thing again

File: src/api.ts [hunk 2]
```diff
@@ -298,11 +294,12 @@ router.post(
       return res.status(400).send(validateBidCancelSchema.errors);
     }
     try {
-      const bidData: Bid | null = await database.getBidBySignedMessage(
+      const bidData: CancelableBid | null = await database.getBidBySignedMessage(
         req.body.bidMessageSignature
       );
 
       if (!bidData) return res.status(400).send("Bid not found");
+      if (bidData.cancelDate && bidData.cancelDate > 0) return res.status(404).send("Bid is no longer active");
```
Comment (RyanCross) [inline src/api.ts:302]: This should be a 400, not a 404. A 404 tells me that the resource at `bid/cancel/{data}` was not found. In this case, it was found, but the bid is no longer active, and thus an invalid request.

**Comment by RyanCross** [comment]
What's the plan for the `bid-archive` collections? Will there be a future script to merge the data from bids-archive into bids?

**Comment by Will-Hyde** [comment]
> e the data from bids-arc
Yep!
Thats the next story, to re-insert cancelled bids back into the bids collection

**Review by JamesEarle — APPROVED** [review]

**Comment by RyanCross** [comment]
> > What's the plan for the bid-archive collections? Will there be a future script to merge the data from bids-archive into bids?
> 
> Yep! Thats the next story, to re-insert cancelled bids back into the bids collection



> > What's the plan for the bid-archive collections? Will there be a future script to merge the data from bids-archive into bids?
> 
> Yep! Thats the next story, to re-insert cancelled bids back into the bids collection

Awesome.

**Review by RyanCross — APPROVED** [review]

**Pull request merged**